controllers/post-controller.js

```diff
@@ -21,4 +26,32 @@ const findPostByID = (req, res) => {
   res.send(foundPost);
 }
 
+const updatePost = (req, res) => {
+  const postToUpdate = req.params.pid;
+  const updatedPost = req.body;
+  const postIndexToUpdate = posts.findIndex((element) => element._id == postToUpdate);
+  posts[postIndexToUpdate] = updatedPost;
```

Comment:
one liner option would be     posts = posts.map(p => p._id === postIdToUpdate ? updatedPost : p);